planner/core/optimizer.go

```diff
@@ -1133,11 +1148,27 @@ func logicalOptimize(ctx context.Context, flag uint64, logic LogicalPlan) (Logic
 			continue
 		}
 		opt.appendBeforeRuleOptimize(i, rule.name(), logic)
-		logic, err = rule.optimize(ctx, logic, opt)
+		var planChanged bool
+		logic, planChanged, err = rule.optimize(ctx, logic, opt)
+		if err != nil {
+			return nil, err
+		}
+		// Compute interaction rules that should be optimized again
+		interactionRule, ok := optInteractionRuleList[rule]
+		if planChanged && ok && isLogicalRuleDisabled(interactionRule) {
+			againRuleList = append(againRuleList, interactionRule)
+		}
+	}
+
+	// Trigger the interaction rule
+	for i, rule := range againRuleList {
+		opt.appendBeforeRuleOptimize(i, rule.name(), logic)
+		logic, _, err = rule.optimize(ctx, logic, opt)
```

Comment:
Should we check the `planChanged` again? And then trigger these interaction rules again until there is no change?